code/modules/economy/quikpay.dm

```diff
@@ -0,0 +1,245 @@
+	var/datum/money_account/quickpay_account = SSeconomy.get_department_account(account)
+	if (!customer_account)
+		to_chat(user, span("notice", "Unable to access account, please contact the Head of Personnel."))
+		return 0
+
+	if(sum == 0)
+		to_chat(user, span("notice", "You need to select something and switch to the confirmation screen before you can pay."))
+		return 0
+
+	if(customer_account.suspended)
+		to_chat(user, span("notice", "Account Suspended, please contact the Head of Personnel."))
+		return 0
+
+	if(customer_account.security_level > 1) //If card requires pin authentication (ie seclevel 1 or 2)
+		var/attempt_pin = input("Enter pin code", "Quik-Pay transaction") as num
+		customer_account = SSeconomy.attempt_account_access(I.associated_account_number, attempt_pin, 2)
+
+		if(!customer_account)
+			to_chat(user, span("notice", "Unable to access account, please contact the Head of Personnel."))
+			return 0
+
+	if(sum > customer_account.money)
+		to_chat(user, span("notice", "Your account lack's the funds to process this payment."))
+		return 0
+	else
+
+
+		// debit money from the purchaser's account
+		customer_account.money -= sum
+		quickpay_account.money += sum
+
+
+
+
+		// create account log in the customers
+		var/datum/transaction/T = new()
+		T.target_name = machine_id
+		T.purpose = "Quik Pay Transaction"
+		T.amount = sum
+		T.source_terminal = machine_id
+		T.date = worlddate2text()
+		T.time = worldtime2text()
+		SSeconomy.add_transaction_log(customer_account,T)
+
+
+		print_receipt()
+		playsound(loc, 'sound/items/polaroid1.ogg', 50, 1)
+		to_chat(user, span("notice", "Transaction to [account] account approved for a total of [sum] credits."))
+
+		sum = 0
+		receipt = ""
+
+// VUEUI Below <3
+
+
+/obj/item/device/nanoquikpay/attack_self(mob/user as mob)
+	var/datum/vueui/ui = SSvueui.get_open_ui(usr, src)
+	if (!ui)
+		ui = new(usr, src, "quikpay-main", 400, 400, "NT Quik-Pay")
+	ui.open()
+
+/obj/item/device/nanoquikpay/vueui_data_change(var/list/data, var/mob/user, var/datum/vueui/ui, var/list/newdata)
+	if(!data)
+		. = data = list()
+
+	VUEUI_SET_CHECK_IFNOTSET(data["items"], items, ., data)
+	VUEUI_SET_CHECK_IFNOTSET(data["price"], items, ., data)
+	VUEUI_SET_CHECK_IFNOTSET(data["tmp_name"], "", ., data)
+	VUEUI_SET_CHECK_IFNOTSET(data["tmp_price"], 0, ., data)
+	VUEUI_SET_CHECK_IFNOTSET(data["_accounts"], station_accounts, ., data)
+	VUEUI_SET_CHECK_IFNOTSET(data["account"], account, ., data)
+	VUEUI_SET_CHECK(data["tmp_price"], max(0, data["tmp_price"]), ., data)
+	if(data["tmp_price"] < 0)
+		data["tmp_price"] = 0
+		. = data
+	VUEUI_SET_CHECK_IFNOTSET(data["selection"], list("_" = 0), ., data)
+	VUEUI_SET_CHECK(data["editmode"], editmode, ., data)
+
+/obj/item/device/nanoquikpay/Topic(href, href_list)
+	var/datum/vueui/ui = href_list["vueui"]
+	if(!istype(ui))
+		return
+	if(href_list["add"])
+
+		if(editmode == 0)
```

Comment:
```suggestion
		if(!editmode)
```

Reply:
Semantically these carry a different connotation. You would do `if (!editmode)` only if you want to test for the presence or absence of it, as if it were an object or a boolean. However, since this is more akin to an enumeration, even if with just 2 members (0 and 1), this check is more appropriate.